### PR TITLE
[office-js] [office-js-preview] Outlook preview updates

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by ElizabethSamuel-MSFT** [comment]
Please do not merge yet as changes are being reviewed. Thanks.

File: types/office-js/index.d.ts [hunk 1]
```diff
@@ -7713,6 +7713,31 @@ declare namespace Office {
...
+         */
+        enum LocationType {
+            /**
+             * A custom location.
+             */
+            Custom = "custom",
+            /**
+             * A conference room or similar resource.
+             */
+            ConferenceRoom = "conferenceRoom"
```
Comment (exextoc) [inline types/office-js/index.d.ts]: This got changed to just room

File: types/office-js/index.d.ts [hunk 2]
```diff
@@ -9367,6 +9392,173 @@ declare namespace Office {
...
+         * 
+         * @remarks
+         * <table>
+         *   <tr>
+         *     <td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td>
+         *     <td>ReadItem</td>
+         *   </tr>
+         *   <tr>
+         *     <td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td>
+         *     <td>Compose</td>
```
Comment (exextoc) [inline types/office-js/index.d.ts]: Compose/read

File: types/office-js/index.d.ts [hunk 3]
```diff
@@ -9367,6 +9392,173 @@ declare namespace Office {
...
```
Comment (exextoc) [inline types/office-js/index.d.ts]: compose/read

File: types/office-js/index.d.ts [hunk 4]
```diff
@@ -9632,6 +9824,54 @@ declare namespace Office {
...
+     * 
+     * [Api set: Mailbox Preview]
+     * 
+     * @beta
+     */
+    export interface LocationDetails {
+        /**
+         * The location's id.
+         */
+        locationId: LocationIdentifier;
```
Comment (exextoc) [inline types/office-js/index.d.ts]: locationIdentifier

File: types/office-js/index.d.ts [hunk 5]
```diff
@@ -9632,6 +9824,54 @@ declare namespace Office {
...
+         */
+        displayName: string;
+        /**
+         * The email address associated with the location.
+         */
+        emailAddress: string;
+        /**
+         * Additional properties for the location.
+         */
+        properties: any;
```
Comment (exextoc) [inline types/office-js/index.d.ts]: We do not return this currently. Maybe we should not document it. Let's confirm from Petra as well and do the needful

File: types/office-js/index.d.ts (hunk 1 above)
Comment (exextoc) [reply types/office-js/index.d.ts]: Room: "room"

File: types/office-js/index.d.ts [hunk 6]
```diff
@@ -9367,6 +9392,173 @@ declare namespace Office {
...
+         *     <td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td>
+         *     <td>Compose</td>
+         *   </tr>
+         * </table>
+         * 
+         * In addition to this signature, this method also has the following signatures:
+         * 
+         * `addAsync(locationIds: LocationIdentifier[], callback?: (result: Office.AsyncResult<LocationDetails[]>) => void): void;`
+         * 
+         * @param locationIds The locations to be added to the current list of locations.
```
Comment (exextoc) [inline types/office-js/index.d.ts]: locationidentifiers

Comment (exextoc) [reply types/office-js/index.d.ts]: locationIdentifier

File: types/office-js/index.d.ts [hunk 7]
```diff
@@ -9733,6 +9973,28 @@ declare namespace Office {
...
+         *   </tr>
+         *   <tr>
+         *     <td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td>
+         *     <td>Appointment Organizer</td>
+         *   </tr>
+         * </table>
+         * 
+         * @beta
+         */
+        enhancedLocations: EnhancedLocation;
```
Comment (exextoc) [inline types/office-js/index.d.ts]: Office.context.mailbox.item.enhancedLocation.add/remove/getAsync()

File: types/office-js/index.d.ts [hunk 8]
```diff
@@ -10820,6 +11082,27 @@ declare namespace Office {
...
+         *   </tr>
+         *   <tr>
+         *     <td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td>
+         *     <td>Appointment Attendee</td>
+         *   </tr>
+         * </table>
+         * 
+         * @beta
+         */
+        enhancedLocations: LocationDetails[];
```
Comment (exextoc) [inline types/office-js/index.d.ts]: enhancedLocation

**Review by exextoc — CHANGES_REQUESTED** [review]
Proposed changes

**Review by exextoc — APPROVED** [review]

**Review by kbrandl — APPROVED** [review]
Verbiage looks good to me. @ElizabethSamuel-MSFT I assume you've verified that syntax of this new content in the d.ts file is correct (by running the autogenerate-docs process with this modified d.ts file and viewing the output docs in a branch on the staging site).

**Comment by ElizabethSamuel-MSFT** [comment]
@kbrandl: Yes, I did that. Thanks.

**Pull request merged**